### PR TITLE
AP_WindVane: Optionalize the WINDVANE device

### PR DESCRIPTION
**Pull request by muramura**
Optionalize the WINDVANE device.
Not all WINDVANE devices need to be built in.
I want to implement only the devices I use.

### CODE REVIEW

File: libraries/AP_WindVane/AP_WindVane_config.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <AP_HAL/AP_HAL_Boards.h>
+
+#ifndef AP_WINDVANE_ENABLED
+#define AP_WINDVANE_ENABLED 1
+#endif
+
+#ifndef AP_WINDVANE_SITL_ENABLED
+#define AP_WINDVANE_SITL_ENABLED AP_WINDVANE_ENABLED
```
Comment (magicrub) [inline libraries/AP_WindVane/AP_WindVane_config.h:10]: Wouldn't the ```CONFIG_HAL_BOARD == HAL_BOARD_SITL``` check make sense here?

File: libraries/AP_WindVane/AP_WindVane_config.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+#ifndef AP_WINDVANE_HOME_ENABLED
+#define AP_WINDVANE_HOME_ENABLED AP_WINDVANE_ENABLED
+#endif
+
+#ifndef AP_WINDVANE_MODERNDEVICE_ENABLED
+#define AP_WINDVANE_MODERNDEVICE_ENABLED AP_WINDVANE_ENABLED
+#endif
+
+#ifndef AP_WINDVANE_BACKEND_ENABLED
+#define AP_WINDVANE_BACKEND_ENABLED AP_WINDVANE_ENABLED
```
Comment (magicrub) [inline libraries/AP_WindVane/AP_WindVane_config.h:30]: AP_WINDVANE_BACKEND_ENABLED should be move up and all the other libraries should check if backend is enabled, not the frontend library. See [AP_Baro_config.h](https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Baro/AP_Baro_config.h) as an example.

**Review by magicrub — CHANGES_REQUESTED** [review]

File: libraries/AP_WindVane/AP_WindVane_config.h (hunk 1 above)
Comment (magicrub) [reply libraries/AP_WindVane/AP_WindVane_config.h:10]: Actually, it looks like we're migrating to use AP_SIM_ENABLED so use that here.

**Review by Hwurzburg — CHANGES_REQUESTED** [review]
needs entries in build_options.py and extract_features.py

**Comment by peterbarker** [comment]
Sorry, looks like this has already be done in master.

Thanks anyway!

**Pull request closed without merging**